RecoLocalMuon/GEMCSCSegment/plugins/GEMCSCSegmentProducer.cc

```diff
@@ -37,12 +38,18 @@ void GEMCSCSegmentProducer::produce(edm::Event& ev, const edm::EventSetup& setup
   LogDebug("GEMCSCSegment") << "start producing segments for " << ++iev << "th event w/ gem and csc data";
 
   // find the geometry (& conditions?) for this event & cache it in the builder
-  edm::ESHandle<CSCGeometry> cscg;
-  setup.get<MuonGeometryRecord>().get(cscg);
+  const edm::ESHandle<CSCGeometry>&& cscg = setup.getHandle(kCSCGeometryToken_);
```

Comment:
```suggestion
  const auto cscg = setup.getHandle(kCSCGeometryToken_);
```
perhaps, here and below? Any reason to use the rvalue reference (&&) here?

Reply:
Hi @jpata, The rvalue reference is my misuse. I fixed it. Thanks!

Reply:
I have one minor question.  In this case, the result of type deduction is `const edm::ESHandle<CSCGeometry>&`, right?

Reply:
It will be `const edm::ESHandle<GEMGeometry>` with the current compiler setup. Copying an `edm::Handle` is cheap AFAIK. When doing a `git grep getHandle` in the base CMSSW repo, you can see that either `const auto h = es.getHandle` or `const auto& h = es.getHandle` are used.

You can use https://diego.assencio.com/?index=05e466a9a1e6dfc958769f80bf986f65 as a recipe to check the result of the type deduction.

Reply:
Thank you so much for your detailed explanation!